keanu-project/src/main/java/io/improbable/keanu/tensor/INDArrayShim.java

```diff
@@ -99,6 +165,231 @@ private static INDArray broadcastMinus(INDArray a, INDArray b) {
         }
     }
 
+    public static INDArray pow(INDArray left, INDArray right) {
```

Comment:
we can do more refactoring if we pass lamdba functions as arguments.

@gordoncaleb might not be keen on the idea though, for performance?

Reply:
Yup, I think that's exactly what I wanted to do and what Caleb talked me out of. We can have another chat about it though?

Reply:
Just shortened it a little bit.